tests/frontend/test_blog.py

```diff
@@ -0,0 +1,140 @@
+    article_page = ArticlePage(selenium, base_url)
+    assert article_title.lower() in article_page.title.text.lower()
+
+
+@pytest.mark.nondesstructive
+def test_open_article_by_clicking_read_more_link(base_url, selenium):
+    page = BlogHomepage(selenium, base_url).open().wait_for_page_to_load()
+    article_title = page.articles[0].title.text
+    page.articles[0].read_more_link.click()
+    article_page = ArticlePage(selenium, base_url)
+    assert article_title.lower() in article_page.title.text.lower()
+
+
+@pytest.mark.nondestructive
+def test_article_page_loaded_correctly(base_url, selenium, variables):
+    selenium.get(f'{base_url}/blog/{variables["blog_article"]}/')
+    page = ArticlePage(selenium, base_url)
+    assert page.header_logo
+    assert page.title
+    assert page.author.name
+    assert page.author.picture
+    assert page.author.twitter_link
+    assert page.author.pocket_link
+    for link in page.nav_bar_links:
+        assert link.is_displayed()
+    # verify that the article has at least one paragraph
+    assert page.content_paragraphs[0].is_displayed()
+    # verify that at least one of next/prev article links is displayed
+    try:
+        assert page.next_article.is_displayed()
+    except NoSuchElementException:
+        page.previous_article.is_displayed()
+
+
+@pytest.mark.nondestructive
+def test_article_page_header_logo_button(base_url, selenium, variables):
+    selenium.get(f'{base_url}/blog/{variables["blog_article"]}/')
+    page = ArticlePage(selenium, base_url)
+    page.header_logo.click()
+    homepage = Home(selenium, base_url)
+    assert homepage.primary_hero.is_displayed()
+
+
+@pytest.mark.nondestructive
+def test_navbar_frontend_homepage_link(base_url, selenium, variables):
+    selenium.get(f'{base_url}/blog/{variables["blog_article"]}/')
+    page = ArticlePage(selenium, base_url)
+    page.nav_bar_links[0].click()
+    homepage = Home(selenium, base_url)
+    assert homepage.primary_hero.is_displayed()
+
+
+@pytest.mark.nondestructive
+def test_navbar_blog_homepage_link(base_url, selenium, variables):
+    selenium.get(f'{base_url}/blog/{variables["blog_article"]}/')
+    page = ArticlePage(selenium, base_url)
+    page.nav_bar_links[1].click()
+    blog_homepage = BlogHomepage(selenium, base_url)
+    assert blog_homepage.articles[0].title.is_displayed()
+
+
+@pytest.mark.nondestructive
+def test_navbar_current_article_link(base_url, selenium, variables):
+    selenium.get(f'{base_url}/blog/{variables["blog_article"]}/')
+    page = ArticlePage(selenium, base_url)
+    page.nav_bar_links[2].click()
+    assert selenium.current_url in f'{base_url}/blog/{variables["blog_article"]}/'
+
+
+@pytest.mark.nondestructive
+def test_next_and_previous_article_links(base_url, selenium, variables):
+    selenium.get(f'{base_url}/blog/{variables["blog_article"]}/')
+    page = ArticlePage(selenium, base_url)
+    initial_title = page.title.text
+    page.next_article.click()
```

Comment:
Maybe we should add a check here that the current title doesn't match the initial title, to make sure that the article pages have been switched.

Reply:
Ok